src/lib.rs

```diff
@@ -169,48 +170,39 @@ pub fn filter_python_executables(
             continue;
         }
         let version_part = &file_name["python".len()..];
-        if let Ok(found_version) = RequestedVersion::from_string(&version_part.to_string()) {
+        if let Ok(found_version) = RequestedVersion::from_str(version_part) {
             match found_version {
-                RequestedVersion::Exact(major, minor) => executables.insert(
-                    Version {
-                        major: major,
-                        minor: minor,
-                    },
-                    path.clone(),
-                ),
+                RequestedVersion::Exact(major, minor) => {
+                    executables.insert(Version { major, minor }, path.clone())
+                }
                 _ => continue,
             };
         }
     }
 
-    return executables;
+    executables
 }
 
 /// Finds the executable representing the latest Python version.
-pub fn choose_executable(
-    version_paths: &collections::HashMap<Version, path::PathBuf>,
-) -> Option<path::PathBuf> {
-    let mut pairs: Vec<(&Version, &path::PathBuf)> = version_paths.iter().collect();
+pub fn choose_executable(version_paths: &HashMap<Version, PathBuf>) -> Option<PathBuf> {
+    let mut pairs: Vec<(&Version, &PathBuf)> = version_paths.iter().collect();
     pairs.sort_unstable_by_key(|p| p.0);
-    match pairs.last() {
-        Some((_, path)) => Some(path.to_path_buf()),
-        None => None,
-    }
+    pairs.last().map(|(_, path)| path.to_path_buf())
 }
 
 /// Checks the environment variable `env_var_name` for a Python version.
-fn check_env_var(env_var_name: &String) -> Result<RequestedVersion, String> {
+fn check_env_var(env_var_name: &str) -> Result<RequestedVersion, String> {
```

Comment:
I take it this is like the `Path` change for `directory_contents()` where `str` not being `Sized` isn't an issue because it's an argument and not a return value? (I suspect you noticed that bumped into the non-constant-sized compile error for returned values a lot, so I just made everything sized. 😁 )

Reply:
It's usually a good idea to receive `&str`. Then you can pass it more kinds of strings, without needing them to be owned (a literal, for example). `&String` can be coerced to `&str`, so it just works great!